feat(anta.cli): Add anta get tests (#843)
aristanetworks/anta: docs/cli/get-tests.md

---
anta_title: Retrieving Tests information
---
<!--
  ~ Copyright (c) 2023-2024 Arista Networks, Inc.
  ~ Use of this source code is governed by the Apache License 2.0
  ~ that can be found in the LICENSE file.
  -->

> `anta get tests` commands allow to list the tests available to ANTA.

Reviewer: ```suggestion
`anta get tests` commands help you discover available tests in ANTA.
```

### Command overview

```bash
Usage: anta get tests [OPTIONS]

  Show all builtin ANTA tests with an example output retrieved from each test
  documentation.

Options:
  --module TEXT  Filter tests by module name.  [default: anta.tests]
  --test TEXT    Filter by specific test name. If module is specified,
                 searches only within that module.
  --short        Display test names without their inputs.
  --count        Print only the number of tests found.
  --help         Show this message and exit.
```

!!! tip

>     By default `anta get tests` will retrieve the tests available in ANTA

Reviewer: ```suggestion
    By default, `anta get tests` will retrieve all tests available in ANTA.
```

        - dot1x
  - VerifyAcctDefaultMethods:
      # Verifies the AAA accounting default method lists for different accounting types (system, exec, commands, dot1x).
      methods:
        - local
        - none
        - logging
      types:
        - system
        - exec
        - commands
        - dot1x
[...]
```

#### Module usage

To retrieve all the tests from `anta.tests.stun`.

``` yaml title="anta get tests --module anta.tests.stun"
anta.tests.stun:
  - VerifyStunClient:
      # Verifies STUN client settings, including local IP/port and optionally public IP/port.
      stun_clients:
        - source_address: 172.18.3.2
          public_address: 172.18.3.21
          source_port: 4500
          public_port: 6006
        - source_address: 100.64.3.2
          public_address: 100.64.3.21
          source_port: 4500
          public_port: 6006
  - VerifyStunServer:
      # Verifies the STUN server status is enabled and running.
```

#### Test usage

``` yaml title="anta get tests --test VerifyTacacsSourceIntf"
anta.tests.aaa:
  - VerifyTacacsSourceIntf:
      # Verifies TACACS source-interface for a specified VRF.
      intf: Management0
      vrf: MGMT
```

!!! tip

>     The test check is using `startswith` in python so you can retrieve all the tests starting with a certain string

Reviewer: ```suggestion
    You can filter tests by providing a prefix - ANTA will return all tests that start with your specified string.
```
